### ISSUE TITLE
我有一个模型代码， 用paddle2.4 运行没有问题， 用paddle2.0运行也能成功，但是梯度都是0，求解

### ISSUE THREAD

**Issue by anygoanygogo** [PaddlePaddle/Paddle#59962]
### 请提出你的问题 Please ask your question

我有一个模型代码， 用paddle2.4 运行没有问题， 用paddle2.0运行也能成功，但是梯度都是0，求解

model = AdsSort(max_seq_len, c_inputs, c_embed, n_heads, c_hidden, n_layers)
loss_fn = nn.CrossEntropyLoss(ignore_index=0)
optimizer = optim.Adam(learning_rate=learning_rate, parameters=model.parameters())

for (batch)
        optimizer.clear_grad()
        outputs = model(inputs)
         loss.backward()
        optimizer.step()
        for name, param in model.state_dict().items():
            if param.grad is not None:
                print(name, param.grad)

**Comment by JZ-LIANG** [comment]
我本地用 2.0 无法复现你的问题， 感觉可能还有其他环境配置因素影响，如果新版本paddle 能够运行成功 ，推荐使用更高版本 paddle
